upload/library/Sidane/Threadmarks/DataWriter/Post.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+class Sidane_Threadmarks_DataWriter_Post extends XFCP_Sidane_Threadmarks_DataWriter_Post
+{
+  protected function _messagePostSave()
+  {
+    parent::_messagePostSave();
+                        
+    if ($this->isChanged('message_state'))
+    {
+      $threadmark = $this->_getThreadMarkModel()->getByThreadIdAndPostId($this->get('thread_id'), $this->get('post_id'));
+                
+      if (!empty($threadmark))
+      {
+        if ($this->get('message_state') == 'visible')
+        {
+          $this->_getThreadMarkModel()->modifyThreadMarkCount($this->get('thread_id'), 1);
+        }
+        else if ($this->isUpdate() && $this->getExisting('message_state') == 'visible')
+        {
+          $this->_getThreadMarkModel()->modifyThreadMarkCount($this->get('thread_id'), -1);
+        }
+      }
+    }        
+  }
+    
+  protected function _messagePostDelete()
+  {
```

Comment:
I think this method needs to call `parent::_messagePostDelete();` as it extends `XenForo_DataWriter_DiscussionMessage_Post` which implements its own `_messagePostDelete` method.